### DISCUSSION TITLE
0.4.1b

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/yi-hack-v5#300] [Announcements]
## What's Changed in `0.4.1b`
* Merging tested commits by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/299

Changes by @roleoroleo 
- [Change size of fifo queues](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/ee0da9d0eb8afdb560953dca9d7f9ac2631d8f92)
- [Restore audio stream ch0_2.h264](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/66030f5abbbd86f186907809cb51c0a3a0af898f)
- [Add native api to take low res snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/cf13502b466fd084cfa7d442ca128056ab7540e8)
- [Remove option: watermark for onvif snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/15e07ff6e9e2d5d39af4fb8da58cb7c325cf32cb)
- [Change label for "Force low res snapshot"](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/c821423f61da34d030435b4081c5290e9221b5f4)
- [Improve RTSP server](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/54b1e653d8aa6136537568b3d8a4b8d7b66bb7cb)
- [Minor fixes](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/e6ebbce764ecd559ec5d612619caba1f9197b154)
- [Remove unused options when empty password](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/9c4fff63880b10779c4a4802f22503fb29cd61d6)
- [Add mqttv4 to watchdog](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/434db6d2ac1c70da9d7895087fcbf48f7fe562a5)
- [Add "if supported" in camera settings page](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/3cedddd5ab5d40d8ea432a1ae7869b4ed6a3f541)
- [Add support for native time osd](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/b305a15a5cf5b99cec55f03058e7d8635c731232)

Changes by @alienatedsec 
- [[Change] launch.sh optimised to kill gently](https://github.com/alienatedsec/yi-hack-v5/commit/63213ca4b62245b752d05ecb25d957f983c986f0) 
- [[Change] Tuning launch.sh](https://github.com/alienatedsec/yi-hack-v5/commit/26009c1e3c4e2fbe635ea7f601a1b7776c7e85c9) 
- [[New] Compile specific module - thanks](https://github.com/alienatedsec/yi-hack-v5/commit/db856f3dbde9e091cd52cf74411d363c39203347) @thelittlefireman

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0...0.4.1b

## What's Changed in `0.4.1a`
* Roleo testing merge for pre-release by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/289

Changes by @roleoroleo 
* [Restore feature to edit config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9a8871666fe542898cce1cc3c0dd5d5e5dd36e6f)
* [Add flag to decrease binary size](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/5c0e1f6d39f53edc84593ff1727380437532e75b)
* [Remove old patch](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/381b840bf043e8c02331fb0d95ff4e19e679093e)
* [Decrease RTP packet length](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/f5e97ce30d01cf6b41081c7f2a4ca3741f19bab9)
* [Improve sync/async read](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9897afc09df6e08870401341e53262bc5a3be558)
* [Update gitignore](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d29169e3cceb7765e9ffb182a3663b3ec51bfc0b)
* [New features: disable, force low and already running](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/70a78bd9ba275af44f93c4d11928b92af4a6d87c)
* [Remove unused settings from config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d19971d4933f1addc75192ad786e3a120ba02729)
* [Add options to disable snapshot and force low res](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/c7034e24c88b105183cdb1b802ec9d4445b6d963)
* [Remove unused mqtt options](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/fcb3bc313482d30c3293fb8a5830f678ba4ba565)
* [Fix and improve h264grabber](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/2dc4cfa895413867f7bdc3b71e594c322d47401c)
* [Fix system.conf parameters](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/6ce47e243809f55bf4df60f9f5b5eadec930b2ad)
* [Remove ai_human_detection and restore baby_crying](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/a309b2aac774f2cb39f34969d7721f9c6df2a02a)

Changes by @alienatedsec 
* [Wi-Fi Recovery Partition](https://github.com/alienatedsec/yi-hack-v5/commit/25db09f312d3f05f5ed64c81ec9f4ad5a5167f64) - Instructions to be issued soon.

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0...0.4.1a

<hr /><em>This discussion was created from the release <a href='https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1b'>0.4.1b</a>.</em>

**Comment by kknishev** [comment]
Hi, missing 'r' in MQTT/Home Assistant/Device Manufactu_er

**Reply by alienatedsec** [reply]
@kknishev sorry but not sure where is that spelling issue. Please can you provide more info?

**Reply by kknishev** [reply]
![image](https://github.com/alienatedsec/yi-hack-v5/assets/22073966/4167aa9b-063c-4258-bc26-6c178c30a7fd)

**Comment by g1za** [comment]
If anybody is having issues in upgrading (can't see the release in the camera interface) this is a workaround: https://github.com/alienatedsec/yi-hack-v5/issues/301#issuecomment-1748287999

**Comment by waleed-alharthi** [comment]
Not sure if it's just me. 0.4.1a initially worked (and the streams were much smoother) but eventually put all my cameras in a boot loop. Manually upgrading the SD to 0.4.1b didn't help.

My config is no cloud, no local recording, just both rtsp streams and mqtt.

I have dome 1080p and outdoor.

**Reply by alienatedsec** [reply]
@waleed-alharthi Boot loop should be fixed with `0.4.1c`
https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1c

**Reply by waleed-alharthi** [reply]
Thanks! I'll try it out today and confirm in a week or so if all keep running ok.

One really dumb thing I did was reset one of the cameras using the button in the back.

I tried to reconfigure WiFi using the vanilla app. Using multiple yi hack v5, v4, and vanilla images. Every time the camera says "connected to WiFi" after the QR but the app keeps waiting for it until it times out. Any hints?

**Reply by alienatedsec** [reply]
@waleed-alharthi That will be your firewall or DNS blocking the camera from connecting to the YI cloud.

**Reply by waleed-alharthi** [reply]
Holy crap! I totally forgot! Thanks!

**Reply by waleed-alharthi** [reply]
I got a camera up and running fine. Others immediately go back to boot loop. I even manage to ping those cameras for a few seconds before they reboot on their own.
Do I need to flush the whole thing with 0.3.8 before upgrading again, or is it ok to use "yi-hack-v5" folder 0.4.1c up from any older version including vanilla or "yi-hack-v4"?

What I've been doing is simply dumping those files into the cameras' SD every time:
[yi-hack-v5] latest
[home_h20] 0.3.8
[rootfs_h20] 0.3.8

![image](https://github.com/alienatedsec/yi-hack-v5/assets/71645462/1dd75156-f319-4a28-83ff-96f6c952182b)

![image](https://github.com/alienatedsec/yi-hack-v5/assets/71645462/4d8a2270-b8d0-4435-9ff1-5f93291c499f)

**Reply by alienatedsec** [reply]
@waleed-alharthi You need to load `0.3.8` of the `yi-hack-v5` before any later versions.

Please beware it's not going to reflash with the same `0.3.8` version over and over as it's a one-time process until you reflash with the other version e.g., `0.3.7` or earlier.

`0.4.1c` has a new WiFi connectivity module to prevent bootloops hopefully.

**Reply by waleed-alharthi** [reply]
This is a useful point. I'll fall back to 0.3.7 then try to go up from there.

**Reply by alienatedsec** [reply]
@waleed-alharthi better to follow this

https://github.com/alienatedsec/yi-hack-v5/discussions/291#discussioncomment-6837396

**Reply by waleed-alharthi** [reply]
Flashed 4 cameras successfully! Many thanks!

**Reply by waleed-alharthi** [reply]
Alrighty, so the report is in.
4 Yi Dome 1080p Cameras flashed
1 Yi Outdoor Camera flashed

Followed the steps:
1. flash 0.3.7
2. flash 0.3.8
3. delete the "yi-hack" folder for 0.3.8 and copy the "yi-hack" folder for 0.4.1c, i left the other 2 files untouched.

All cameras worked fine on day 1.
I could see the rtsp streams on frigate and all was well.

Within 1 week:
- 3 Yi Dome 1080p Cameras went into boot loop.
- 1 Yi Outdoor camera went into boot loop.
- 1 Yi Dome 1080p is working well!

I have done something interesting, and It may be the cause of my pain.
I initially configured all the settings using the Web UI on that one camera which is still working today.
I exported the config, and imported it into all the other cams. I simply changed the "Hostname" and the MQTT Client ID. Could this be the issue?
